### PR TITLE
feat(adonet): added support for ado.net based db instrumentation

### PR DESCRIPTION
**Pull request by tvaintrob**

### CODE REVIEW

**Comment by henprtz** [comment]
Let's check that we support exceptions coming during the execution of the request and also that we handle errors and show error returned from the query itself - meaning query failed or something of that sort. @tvaintrob

File: src/Epsagon.Dotnet.Instrumentation/ADONET/EpsagonTracedCommand.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            using (var scope = GlobalTracer.Instance
+                            .BuildSpan("")
+                            .WithStartTimestamp(DateTime.Now)
+                            .StartActive(finishSpanOnDispose: true))
+            {
+                var span = scope.Span;
+                object result = null;
+
+                SpanDefaults(span);
+                span.SetTag("sql.table_name", ""); // check other instrum to see how this is obtained
```
Comment (henprtz) [inline src/Epsagon.Dotnet.Instrumentation/ADONET/EpsagonTracedCommand.cs:86]: Can we will this info from the query itself / connection object?

File: src/Epsagon.Dotnet.Instrumentation/ADONET/EpsagonTracedCommand.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            span.SetTag("resource.name", DbConnection.Database);
+            span.SetTag("resource.operation", operation);
+            span.SetTag("sql.driver", DbConnection.GetType().FullName);
+            span.SetTag("sql.statement", CommandText);
+            span.SetTag("sql.connection_string", DbConnection.ConnectionString);
+        }
+
+        public override int ExecuteNonQuery()
+        {
+            using (var scope = GlobalTracer.Instance
```
Comment (henprtz) [inline src/Epsagon.Dotnet.Instrumentation/ADONET/EpsagonTracedCommand.cs:51]: If this code throws an exception we need to have a failsafe mechanism that will still run the user's code

**Review by henprtz — CHANGES_REQUESTED** [review]
Looks good; Just small comment on the table name & params collection and also on how we prevent our code from breaking the customer's code

File: src/Epsagon.Dotnet.Instrumentation/ADONET/EpsagonTracedCommand.cs (hunk 1 above)
Comment (tvaintrob) [reply src/Epsagon.Dotnet.Instrumentation/ADONET/EpsagonTracedCommand.cs:96]: yeah didnt get to it yet, we need to kindof parse the sql to get it

**Review by henprtz — APPROVED** [review]

**Pull request merged**